### ISSUE TITLE
WebService Importer/Exporter

### ISSUE THREAD

**Issue by phochste** [LibreCat/Catmandu#114]
Create an importer to push JSON to an URI

**Comment by vpeil** [comment]
I have a similar intention for the DataCite metadata store (https://www.datacite.org/), i.e. DOI registry service for research data.

The necessary steps are:
- create XML file in datacite format
- HTTP POST this file together with your username/password

@phochste Should we have another base class, e.g. Catmandu::Exporter::Webservice?
Then we can build new Exporter like Catmandu::Exporter::Webservice::Datacite, etc.

**Comment by phochste** [comment]
@vpeil Yes, but I am a but uncomfortable with the naming convention.  Catmandu::Exporter::Datacite that uses Catmandu::Exporter::Webservice as base class is ok for me. But be aware, I have issues with Catmandu::Exporter::MARC that used Perl subclassing in the past and got to problems having fixes executed twice.

**Comment by vpeil** [comment]
ok, then what about **Catmandu::Webservice** as base class?

On 13/03/15 09:30, Patrick Hochstenbach wrote:
>
> @vpeil <https://github.com/vpeil> Yes, but I am a but uncomfortable 
> with the naming convention. Catmandu::Exporter::Datacite that uses 
> Catmandu::Exporter::Webservice as base class is ok for me. But be 
> aware, I have issues with Catmandu::Exporter::MARC that used Perl 
> subclassing in the past and got to problems having fixes executed twice.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/LibreCat/Catmandu/issues/114#issuecomment-78866375>.
>

-- 
Vitali Peil
Fachreferent
PUB <pub.uni-bielefeld.de>
Raum E1-144, Tel. 0521 106 6125
Universitätsbibliothek Bielefeld

**Comment by phochste** [comment]
Sure. For now all exporters are Catmandu::Exporter::<***> nothing deeper.
